catalyst_voices/lib/pages/discovery/toggle_state_text.dart

```diff
@@ -22,14 +22,28 @@ class _ToggleStateTextState extends State<ToggleStateText> {
   @override
   void initState() {
     super.initState();
-    _tapVisitor.onTap = () {
-      context.read<SessionBloc>().add(const VisitorSessionEvent());
+    _tapVisitor.onTap = () async {
+      await context.read<SessionCubit>().removeKeychain();
     };
-    _tapGuest.onTap = () {
-      context.read<SessionBloc>().add(const GuestSessionEvent());
+    _tapGuest.onTap = () async {
+      final sessionBloc = context.read<SessionCubit>();
+
+      if (sessionBloc.state is ActiveAccountSessionState) {
+        await sessionBloc.lock();
+      }
+
+      if (sessionBloc.state is VisitorSessionState) {
+        await sessionBloc
+            .switchToDummyAccount()
+            .then((_) => sessionBloc.lock());
+      }
```

Comment:
imho we should use else-if to avoid checking further states if we found a match already

```suggestion
      if (sessionBloc.state is ActiveAccountSessionState) {
        await sessionBloc.lock();
      } else if (sessionBloc.state is VisitorSessionState) {
        await sessionBloc
            .switchToDummyAccount()
            .then((_) => sessionBloc.lock());
      }
```